docs/usage/shoot_high_availability_best_practices.md

```diff
@@ -0,0 +1,431 @@
+
+Basically, the `startupProbe` and the `livenessProbe` help you restart the container, if it's unhealthy for whatever reason, by letting the `kubelet` that orchestrates your containers on a node know, that it's unhealthy. The former is a special case of the latter and only applied at the startup of your container, if you need to handle the startup phase differently (e.g. with very slow starting containers) from the rest of the lifetime of the container.
+
+Now, the `readinessProbe` helps you manage the ready status of your container and thereby pod (any container that is not ready turns the pod not ready). This again has impact on endpoints and pod disruption budgets:
+
+- If the pod is not ready, the endpoint will be removed and the pod will not receive traffic anymore
+- If the pod is not ready, the pod counts into the pod disruption budget and if the budget is exceeded, no further voluntary pod disruptions will be permitted for the remaining ready pods (e.g. no eviction, no voluntary horizontal or vertical scaling, if the pod runs on a node that is about to be drained or in draining, draining will be paused until the max drain timeout passes)
+
+As you can see, all of these probes are (also) related to HA (mostly the `readinessProbe`, but depending on your workload, you can also leverage `livenessProbe` and `startupProbe` into your HA strategy). If Kubernetes doesn't know about the individual status of your container/pod, it won't do anything for you (right away). That said, later/indirectly something might/will happen via the node status that can also be ready or not ready, which influences the pods and load balancer listener registration (a not ready node will not receive cluster traffic anymore), but this process is worker pool global and reacts delayed and also doesn't discriminate between the containers/pods on a node.
+
+In addition, Kubernetes also offers [pod readiness gates](https://kubernetes.io/docs/concepts/workloads/pods/pod-lifecycle/#pod-readiness-gate) and some cloud provider cloud-controller-managers support them natively (e.g. [AWS](https://kubernetes-sigs.github.io/aws-load-balancer-controller/v2.1/deploy/pod_readiness_gate)). If you can, make use of them to incorporate also the LB registration state into your overall pod readiness (normally, only the sum of the container readiness matters, but pod readiness gates additionally count into the overall pod readiness). That's the only way, with Kubernetes means, to block (by means of pod disruption budgets that we will talk about next) the roll-out of your workload/nodes in case the LB registration fails (e.g. rate limited or whatever).
```

Comment:
gardener-extension-provider-aws is not using https://github.com/kubernetes-sigs/aws-load-balancer-controller, it is using https://github.com/kubernetes/cloud-provider-aws. Let's remove this link/text or link other CCM that supports the feature. Otherwise, it can be confusing for the reader (docs stated that it is supported on AWS, but it does not work on my cluster).

Reply:
Very good catch, dropped and replaced with a more general introduction of the feature:

> In addition, Kubernetes also offers [pod readiness gates](https://kubernetes.io/docs/concepts/workloads/pods/pod-lifecycle/#pod-readiness-gate) to amend your pod readiness with additional custom conditions (normally, only the sum of the container readiness matters, but pod readiness gates additionally count into the overall pod readiness). This may be useful if you want to block (by means of pod disruption budgets that we will talk about next) the roll-out of your workload/nodes in case some (possibly external) condition fails.